packages/core/src/render3/util.ts

```diff
@@ -146,7 +166,7 @@ export function getRootView(target: LView | {}): LView {
   ngDevMode && assertDefined(target, 'component');
   let lView = Array.isArray(target) ? (target as LView) : readPatchedLView(target) !;
   while (lView && !(lView[FLAGS] & LViewFlags.IsRoot)) {
-    lView = lView[PARENT] !;
+    lView = lView[PARENT] !as LView;
```

Comment:
We may walk up components that are inside embedded views

```suggestion
    lView = getLViewParent(lView) !;
```

Also, this looks almost identical to `getRootView` in `discovery_utils` (which you updated above as well). We should remove one of these as part of the cleanup.

Reply:
Removed the other implementation of `getRootView`, and also a duplicate of `getRootContext`. :man_shrugging: